eelsdbconverter/__init__.py

```diff
@@ -49,60 +44,129 @@ def __init__(self):
-            first_name=' '.join(author.split(' ')[0:-1]),
-            last_name=author.split(' ')[-1])]
-        archive.section_metadata.comment = data.get('description')
-        archive.section_metadata.references = [
-            data.get('permalink'),
-            data.get('api_permalink'),
-            data.get('download_link')
-        ]
-
-        reference = data.get('reference')
-        if reference:
-            if isinstance(reference, str):
-                archive.section_metadata.references.append(reference)
-            elif isinstance(reference, dict):
-                if 'freetext' in reference:
-                    archive.section_metadata.references.append(
-                        re.sub(r'\r+\n+', '; ', reference['freetext']))
-                else:
-                    archive.section_metadata.references.append('; '.join([
-                        str(value).strip() for value in reference.values()
-                    ]))
-
-        section_data.repository_name = 'Electron Energy Loss Spectroscopy (EELS) database'
-        section_data.repository_url = 'https://eelsdb.eu'
-        section_data.entry_repository_url = data.get('permalink')
+            sample.elements = elements
+
+        # Experiment
+        experiment = metadata.m_create(Experiment)
+        experiment.method_name = 'electron energy loss spectroscopy'
+        experiment.method_abbreviation = 'EELS'
+        experiment.experiment_publish_time = datetime.strptime(
+            file_data.get('published'), '%Y-%m-%d %H:%M:%S')
+        edges = file_data.get('edges')
+        if edges is not None:
+            if isinstance(edges, str):
+                edges = json.loads(edges)
+            experiment.edges = edges
+
+        # Instrument
+        instrument = metadata.m_create(Instrument)
+        instrument.source_label = file_data['microscope']
+        device_settings = instrument.m_create(DeviceSettings)
+        device_settings.device_name = file_data['microscope']
+        device_settings.max_energy = file_data['max_energy']
+        if file_data.get('min_energy') is not None:
+            device_settings.min_energy = file_data['min_energy']
+        device_settings.guntype = file_data['guntype']
+        device_settings.beam_energy = file_data['beamenergy']
+        if file_data.get('resolution') is not None:
+            device_settings.resolution = file_data['resolution']
+        device_settings.step_size = file_data['stepSize']
+        if file_data.get('acquisition_mode') is not None:
+            device_settings.acquisition_mode = file_data['acquisition_mode']
+        if file_data.get('beamcurrent') is not None:
+            device_settings.beam_current = file_data['beamcurrent']
+        device_settings.detector_type = file_data['detector']
+        device_settings.dark_current = file_data['darkcurrent']
+
+        # Author Generated
+        author_generated = metadata.m_create(AuthorGenerated)
```

Comment:
Again, the author info should be in a section Author. (For external dbs) This should be a subsection of origin. Other user information (uploader, coauthors are treated by nomad and stored somehwere else). 

The description belongs to section measurement. No AuthorGenerated pls. We can have a category to denote "author generated": 

```python
class UserProvided(metainfo.Categroy):
    pass

class Measurement(metainfo.Section):
    description = Quantity(type=str, categories=[UserProvided]))
```

Reply:
Most sections should get their own UserProvided quantities. Measurement, Sample, Instrument, etc. all should have something like notes, comment?

Reply:
I created a category like you suggested. I agree that it makes more sense to do it this way. This was the desired result of having a section called `AuthorGenerated`. Just check that I've done this correctly.